cpp/src/io/parquet/page_enc.cu

```diff
@@ -1237,6 +1243,35 @@ class header_encoder {
     current_field_index   = field;
   }
 
+  inline __device__ void field_list_begin(int field, size_t len, int type)
+  {
+    current_header_ptr = cpw_put_fldh(current_header_ptr, field, current_field_index, ST_FLD_LIST);
+    current_header_ptr =
+      cpw_put_byte(current_header_ptr, (uint8_t)((std::min(len, (size_t)0xfu) << 4) | type));
```

Comment:
```suggestion
      cpw_put_byte(current_header_ptr, static_cast<uint8_t>((std::min(len, size_t{0xfu} << 4) | type));
```

Reply:
I was wondering about the old style casts.  I was going to change them but noticed they were used in the existing code.  I'll try to squash all the ones I introduced.